src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -309,38 +330,70 @@ def _run_microtrait(genome_id, fna_file, batch_dir):
-        data.update({name: list(genes_detected_table[idx])})
-    genes_detected_df = pd.DataFrame(data=data)
-    genes_detected_df.to_csv(os.path.join(genome_dir, 'genes_detected_table.csv'), index=False)
-
-    # remove the unpacked assembly file to save space
-    if remove_gz_file:
-        print(f'removing {fna_file}')
-        os.remove(fna_file)
+    try:
+        # Load the R script as an R function
+        r_script = """
+        library(microtrait)
+        extract_traits <- function(genome_file, out_dir) {
+          genome_file <- file.path(genome_file)
+          microtrait_result <- extract.traits(in_file = genome_file, out_dir = out_dir)
+          return(microtrait_result)
+        }
+        """
+        r_func = robjects.r(r_script)
+        r_result = r_func(fna_file, genome_dir)
+
+        microtrait_result = _get_r_list_element(r_result, 'microtrait_result')
+
+        trait_counts_atgranularity3 = _get_r_list_element(microtrait_result, 'trait_counts_atgranularity3')
+        trait_counts_atgranularity3_df = _r_table_to_df(trait_counts_atgranularity3)
+
+        trait_counts_atgranularity3_df.to_csv(os.path.join(genome_dir, 'trait_counts_atgranularity3.csv'), index=False)
+
+    except Exception as e:
+        raise ValueError(f'Error running microtrait on {fna_file}') from e
+    finally:
+        # remove the unpacked assembly file to save space
+        if remove_gz_file:
+            print(f'removing {fna_file}')
+            os.remove(fna_file)
+
+    selected_cols = ['microtrait_trait-value', 'microtrait_trait-displaynameshort', 'microtrait_trait-displaynamelong']
+    cell_info_df = trait_counts_atgranularity3_df[selected_cols]
+
+    # Extract the substring of the 'microtrait_trait-displaynamelong' column before the last colon character
+    # and assign it to a new 'category' column in the DataFrame
+    cell_info_df['category'] = cell_info_df['microtrait_trait-displaynamelong'].str.rsplit(':', 1).str[0]
+    # Use the microtrait_trait-displaynameshort column as the column name, microtrait_trait-displaynamelong column
+    # as the column description, and microtrait_trait-value as the cell value
+    cell_info_df = cell_info_df.rename(columns={'microtrait_trait-displaynameshort': 'col_name',
+                                                'microtrait_trait-displaynamelong': 'col_description',
+                                                'microtrait_trait-value': 'cell_val'})
+
+    categories, cells, celid, min_value, max_value = list(), list(), 0, float('inf'), float('-inf')
+    for category_name, category_df in cell_info_df.groupby('category'):
+        columns = list()
+        for i, row in category_df.iterrows():
+            columns.append(ColumnInformation(id=str(i), name=row['col_name'], description=row['col_description']))
+            cell_val = float(row['cell_val'])
+            cells.append(Cell(celid=str(celid), colid=str(i), val=cell_val))
```

Comment:
Every cell in the entire heatmap has to have a unique ID. These IDs are only unique per genome

Reply:
👍 I will handle this in the parser script.

Reply:
:+1: moot now with the removal of this code